x-pack/test/functional/apps/ml/short_tests/settings/calendar_creation.ts

```diff
@@ -96,12 +98,143 @@ export default function ({ getService }: FtrProviderContext) {
+      );
+      await createSingleGroupJobs();
+
+      await ml.testExecution.logTestStep('calendar creation loads the calendar management page');
+      await ml.navigation.navigateToMl();
+      await ml.navigation.navigateToSettings();
+      await ml.settings.navigateToCalendarManagement();
+
+      await ml.testExecution.logTestStep('calendar creation loads the new calendar edit page');
+      await ml.settingsCalendar.assertCreateCalendarButtonEnabled(true);
+      await ml.settingsCalendar.navigateToCalendarCreationPage();
+
+      await ml.settingsCalendar.waitForFormEnabled();
+
+      await ml.testExecution.logTestStep(
+        'calendar creation verifies the job selection and job group section are displayed'
+      );
+      await ml.settingsCalendar.assertJobSelectionExists();
+      await ml.settingsCalendar.assertJobSelectionEnabled(true);
+      await ml.settingsCalendar.assertJobGroupSelectionExists();
+      await ml.settingsCalendar.assertJobGroupSelectionEnabled(true);
+
+      await ml.testExecution.logTestStep('calendar creation sets the calendar id');
+      await ml.settingsCalendar.setCalendarId(calendarId);
+
+      await ml.testExecution.logTestStep('calendar creation sets the job selection');
+      await asyncForEach(jobConfigs, assignJobToCalendar);
+
+      await ml.settingsCalendar.saveCalendar();
+      await ml.settingsCalendar.assertCalendarRowExists(calendarId);
+
+      await ml.testExecution.logTestStep(
+        'Assert that the calendar is only connected to jobs applied during creation'
+      );
+      await assertOnlyConnectedToJobsAppliedDuringCreation();
+
+      await ml.testExecution.logTestStep(
+        'Assert that the calendar can connect to jobs applied after creation'
+      );
+      await assertConnectedToJobsAppliedAfterCreation();
+    });
+
+    async function assignJobToCalendar(
+      jobConfig: ReturnType<typeof createJobConfig>
+    ): Promise<void> {
+      await ml.settingsCalendar.selectJob(jobConfig.job_id);
+    }
+
+    async function createSingleGroupJobs() {
+      const automatedConfig = createJobConfig('test_calendar_ad_3');
+      const multiMetricConfig = createJobConfig('test_calendar_ad_4');
+      automatedConfig.groups = ['automated'];
+      multiMetricConfig.groups = ['multi-metric'];
+
+      // @ts-expect-error not full interface
+      await asyncForEach(
+        [automatedConfig, multiMetricConfig],
+        async (config) => await ml.api.createAnomalyDetectionJob(config)
+      );
+    }
+
+    async function assertOnlyConnectedToJobsAppliedDuringCreation() {
+      // Assert that the Calendar Management view shows that the calendar is only connected to the jobs
+      // applied at creation
+      const groupsJustAfterCreationVisibleText = await testSubjects.getVisibleText(
+        'mlCalendarListColumnJobs'
+      );
+      expect(groupsJustAfterCreationVisibleText).to.match(/test_calendar_ad_1, test_calendar_ad_2/);
```

Comment:
This should be a service method in `ml.settingsCalendar` (where we already have a few helpers to read and interact with the calendars table). It should also do the assertion on the calendar row basis and not on the whole table - basically making sure that a certain calendar entry has the expected job entries, e.g. something like `assertCalendarRowJobs`.

Reply:
[Here we go](https://github.com/elastic/kibana/pull/178553/commits/640b31156f158167030318aec86509223f185cf0) boss!